File: Clases2024-25/Clase12_Persistencia/app/src/main/java/com/example/clase12_persistencia/PersonaDAO.java
package com.example.clase12_persistencia;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public abstract class PersonaDAO {

    @Query("SELECT * FROM persona ORDER BY id ASC")
    public abstract List<Persona> getAll();

    @Insert
    public abstract void insertaPersona(Persona persona );

    @Delete
    public abstract void deletePersona(Persona p);

    @Query("DELETE FROM persona WHERE id = :id")
    public abstract void deletePersona(int id);


    @Update
    public abstract void actualizaPersona(Persona persona);

    @Query("UPDATE persona SET nombre = :nombre, edad = :edad, direccion = :direccion WHERE id = :id")
    public abstract void actualizaPersona2(int id, String nombre, int edad, String direccion);


}

File: Clases2023-24/Clase31_AccesoBaseDatos_Coroutines/app/src/main/java/com/example/clase31_accesobasedatos_coroutines/PersonaDao.kt
package com.example.clase31_accesobasedatos_coroutines

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface PersonaDao {
    //CRUD - Create, Read, Update, Delete
    @Query("SELECT * FROM persona ORDER BY id ASC")
    suspend fun getAll(): MutableList<Persona>

    @Query("SELECT * FROM persona WHERE id = (:personaIds)")
    suspend fun getPersonaById(personaIds: IntArray): Persona

    @Insert
    suspend fun insertPersona(vararg persona: Persona)

    @Delete
    suspend fun deletePersona(persona: Persona)

    /** Opción 1
     * @Query("UPDATE persona SET nombre = :nombre, edad = :edad, direccion = :direccion WHERE id = :id")
     * suspend fun updatePersona(nombre: String, edad: Int, direccion: String, id: Int)
     */


    // Opción 2
    @Update
     suspend fun updatePersona(persona: Persona)
     /*
    @Query("INSERT INTO persona (nombre, edad, direccion) VALUES (:nombre, :edad, :direccion)")
    fun insertPersona(nombre: String, edad: Int, direccion: String)
    */

}